api/envoy/config/filter/http/transformation/v2/transformation_filter.proto

```diff
@@ -153,6 +153,19 @@ message Transformation {
 // Extractions can be used to extract information from the request/response.
 // The extracted information can then be referenced in template fields.
 message Extraction {
+  // The mode of operation for the extraction.
+  enum Mode {
+    // Default mode. Extract the value of the subgroup-th capturing group.
+    EXTRACT = 0; 
+    // Replace the value of the subgroup-th capturing group with the replacement_text.
+    // Note: replacement_text must be set for this mode.
+    SINGLE_REPLACE = 1;
```

Comment:
Shouldnt this act like replace_all

Reply:
Updated `SINGLE_REPLACE` to return the original input if no match is found bd9380cac43b5f488c2161e6923d92aac1577983

I was hesitant about this at first, but this logic changed my mind:
 - in extract mode, a user's goal is to extract a value from a given request. if the regex doesn't match, then we can't extract that value, so we return nothing
 - in the replace modes, a user's goal is to overwrite portions of a given input. if the regex doesn't match, we can't overwrite anything, so we return the original input

I think it's also much easier for users to understand if the new modes have the same behavior when no match is found